### ISSUE TITLE
Selling the same items several times

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#273]
```
It is possible to sell the same times several times, providing an way to cheat 
boost gold.

What steps will reproduce the problem?
1. Go to any shop
2. Open the sell tab
3. Now tap the sell button on the same item as quickly as possible. Keep 
tapping until one second after the dialogue with the number of items to sell 
has appeared.
4. Now sell all of your items in the dialogue that has appeared. When the 
dialog is disposed you will, if you are lucky, have more sell dialogues under 
it.
5. Repeat #4 for all the dialogues
6. You have now gained more money than you were suppost to be earning.

What version of the product are you using? On what device?
0.6.10, ZTE Blade
```

Original issue reported on code.google.com by `Jimmy.Ax...@gmail.com` on 12 Nov 2011 at 6:38

**Comment by GoogleCodeExporter** [comment]
```
Confirmed.

You also may duplicate rare items when buying them back from the shop. 
Unfortunately this does not work with legendary items. :-(

However extraordinary can be reproduced this way...
o.O
```

Original comment by `SamuelPl...@gmail.com` on 13 Nov 2011 at 8:52

**Comment by GoogleCodeExporter** [comment]
Original comment by `sdeva...@gmail.com` on 14 Nov 2011 at 5:12
- Changed state: **Accepted**
- Added labels: **Priority-High**

**Comment by GoogleCodeExporter** [comment]
```
the attached patch fixes the issue for me
```

Original comment by `chi...@gmail.com` on 14 Nov 2011 at 3:17

Attachments:
- [issue273.patch](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-273/comment-3/issue273.patch)

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the code patch chitul, looks great!
Committed in r203.
```

Original comment by `oskar.wi...@gmail.com` on 14 Nov 2011 at 7:49
- Changed state: **Ready**
- Added labels: **Milestone-0.6.11**

**Comment by GoogleCodeExporter** [comment]
```
Released in v0.6.11. Closing issue.
```

Original comment by `oskar.wi...@gmail.com` on 9 Jun 2012 at 9:31
- Changed state: **Fixed**
